### ISSUE TITLE
How to order by a date column

### ISSUE THREAD

**Issue by windgaucho** [GriddleGriddle/Griddle#519]
Hello I want to know how can I sort by a date metadata column? I had readed all issues about sort by date but I can get it to work. Can somebody tell me how to sort by a date column? May be an example? Thanks

**Comment by babenzele** [comment]
Your date object will sort just fine by default, it's the rendering that needs to be custom. You can create a [customComponent](https://griddlegriddle.github.io/Griddle/customization.html) to render your date object using [moment](http://momentjs.com/) or something similar. 

```javascript
import React from 'react';
import moment from 'moment';

class DateCell extends React.Component {
  renderDate() {
    const date = this.props.data;
    if (date) {
      return <span>{moment(date).calendar()}</span>;
    } else {
      return <span></span>;
    }
  }
  render() {
    return <div>{this.renderDate()}</div>;
  }
}

export default DateCell;
```

**Comment by babenzele** [comment]
I've been poking around on date sorting and @windgaucho is right, dates do not sort correctly.

**Comment by windgaucho** [comment]
@babenzele thanks for your response and example!!

I finally made it work using the number representation fo a date value.
My java rest service was returning a date column using the dd/mm/yyyy format, but like I say previously I can not sort by date using this format. So I change the format of my java rest service to return the date values using number representations (like this: http://stackoverflow.com/questions/12067697/convert-current-date-as-integer) and since the date value is now a number the order work as espected.
I am using your example code for render a date value for the column using moment.js to create and format a javascript date from a date number representation:

`import React, { PropTypes } from 'react';
import moment from 'moment';

class DateColumn extends React.Component {
  constructor(props) {
    super(props);
  }
  renderDate() {
    const date = this.props.data;
    if (date) {
      return <span>{moment(new Date(date)).format("DD/MM/YYYY")}</span>;
    } else {
      return <span />;
    }
  }

  render() {
    return (
      <div>{this.renderDate()}</div>
    );
  }
}

DateColumn.propTypes = {
  data: PropTypes.number.isRequired
};

export default DateColumn;`

**Comment by babenzele** [comment]
Ok, so I got this working as well. 

I created a `customCompareFn` for the date column and referenced the function in `columnMetadata`. 

For the function, you can either supply two arguments so that you can compare them yourself, or you can supply one argument (shown here), which you can modify and return so that `orderBy` from the lodash library can properly sort the dates. 

```javascript
const dateSortFn = (a) => {
  return a.toISOString() // 2011-10-05T14:48:00.000Z
}
```

```javascript
{
  columnName: 'updatedAt',
  displayName: 'Last modified',
  customComponent: DateCell,
  customCompareFn: dateSortFn
}
```

https://github.com/GriddleGriddle/Griddle/blob/3c1d4255f379a5d5eb3badae0b2769a03e9b4a14/scripts/griddle.jsx#L558-L571

**Comment by ryanlanciaux** [comment]
Thank you @babenzele! In addition, we've released Griddle 1.0 that takes a custom sort method on column definitions (so a column can be given a sort function to use).

Please [see the new API docs](http://griddlegriddle.github.io/Griddle/docs/api/) under `ColumnDefinition`.`sortMethod`. and [our Storybook story for manually testing this functionality](https://github.com/GriddleGriddle/Griddle/blob/master/stories/index.js#L119).

I'm going to close this one for now but please feel free to follow-up here or open a new issue if any problems are encountered. 👍
